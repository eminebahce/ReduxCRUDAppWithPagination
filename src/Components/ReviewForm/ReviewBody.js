import React, {Component} from 'react'
import {TextField} from "@material-ui/core";

class ReviewBody extends Component {

    render() {
        return (
            <TextField id="formControlsTextarea"
                       rowsMax="4"
                       label="Review:"
                       placeholder="Enter your review here"
                       required={true}
                       multiline
                       data-test="formTextArea"
            />
        );
    }
}

export default ReviewBody;
