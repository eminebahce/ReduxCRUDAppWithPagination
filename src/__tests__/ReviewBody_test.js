import React from 'react';
import {shallow} from 'enzyme';
import ReviewBody from '../../src/Components/ReviewForm/ReviewBody';
import {Provider} from "react-redux";
import store from "../store";
import {TextField} from "@material-ui/core";
import renderer from 'react-test-renderer';

describe('<ReviewBody /> component', () => {
    it('should render without errors', () => {
        const component = shallow(<ReviewBody />);
        console.log(component.debug());
        const wrapper = component.find(`[data-test='formTextArea']`);
        expect(wrapper.length).toBe(1);
    });

    describe('renders ReviewBody element with TextField', () => {
        it('should not return error', () => {
            const component = renderer.create(
                <Provider store={store}>
                    <TextField />
                </Provider>
            );
            let tree = component.toJSON();
            expect(tree).toMatchSnapshot();
        });
    });
});

