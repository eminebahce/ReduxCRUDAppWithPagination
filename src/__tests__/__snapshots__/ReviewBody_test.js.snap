// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ReviewBody /> component renders ReviewBody element with TextField should not return error 1`] = `
<div
  className="MuiFormControl-root-1"
>
  <div
    className="MuiInputBase-root-18 MuiInput-root-5 MuiInput-underline-9 MuiInputBase-formControl-19 MuiInput-formControl-6"
    onClick={[Function]}
  >
    <input
      aria-invalid={false}
      className="MuiInputBase-input-28 MuiInput-input-13"
      disabled={false}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      required={false}
      type="text"
    />
  </div>
</div>
`;

exports[`renders ReviewBody element with TextField should not return error 1`] = `
<div
  className="MuiFormControl-root-1"
>
  <div
    className="MuiInputBase-root-18 MuiInput-root-5 MuiInput-underline-9 MuiInputBase-formControl-19 MuiInput-formControl-6"
    onClick={[Function]}
  >
    <input
      aria-invalid={false}
      className="MuiInputBase-input-28 MuiInput-input-13"
      disabled={false}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      required={false}
      type="text"
    />
  </div>
</div>
`;
